File: app/src/main/java/com/group15/fitnesstracker/db/History.java
package com.group15.fitnesstracker.db;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(primaryKeys = {"userId, workoutId"},foreignKeys = {@ForeignKey(entity = User.class,parentColumns = "id",childColumns = "userId"),
                        @ForeignKey(entity = Workout.class,parentColumns = "workoutId",childColumns = "workoutId")
})
public class History {
    public int userId;

    public int workoutId;
}

File: app/src/main/java/com/group15/fitnesstracker/db/History.kt
package com.group15.fitnesstracker.db

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(foreignKeys = [ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["userId"]),
                       ForeignKey(entity = Workout::class, parentColumns = ["workoutId"], childColumns = ["workoutId"])])
data class History (
    @PrimaryKey(autoGenerate = true) var recordingId: Int = 0,
    var userId: Int = 0,
    var workoutId: Int = 0
)


